### PR TITLE
Improving register rotation example with notes and references

### PR DESCRIPTION
**Pull request by mellertd**
## Description

Follow up on discussion in #4515. I also broke things up a little differently to fix some problems with how the example is rendering on the website.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by alexdesiqueira** [comment]
Thank you @mellertd, this looks great!

**Comment by mellertd** [comment]
Just checking on whether this is ok?

File: doc/examples/registration/plot_register_rotation.py [hunk 1]
```diff
@@ -235,3 +240,41 @@
...
+# applied, particularly to remove high frequencies, and different choices here
+# may impact outcome. The bandpass filter also complicates matters because,
+# since the images to be registered will differ in scale and these scale
+# differences are unknown, any bandpass filter will necessarily attenuate
+# different features between the images. For example, the log-polar transformed
+# magnitude spectra don't really look "alike" in the last example here. Yet if
+# you look closely, there are some common patterns in those spectra, and they
+# do end up aligning well by phase correlation as demonstrated.
+#
+# 2. Images must be windowed using windows with cirular symmetry, to remove the
```
Comment (jni) [inline doc/examples/registration/plot_register_rotation.py:262]:
```suggestion
# 2. Images must be windowed using windows with circular symmetry, to remove the
```

File: doc/examples/registration/plot_register_rotation.py [hunk 2]
```diff
@@ -235,3 +240,41 @@
...
+#
+# Further reading:
+#
+# B.S. Reddy and B.N. Chatterji. An FFT-based technique for translation,
+# rotation and scale- invariant image registration. IEEE Trans. Image
+# Processing, 5(8):1266–1271, 1996.
+#
+# Tzimiropoulos, Georgios, and Tania Stathaki. "Robust FFT-based
+# scale-invariant image registration." In 4th SEAS DTC Technical Conference.
+# 2009.
```
Comment (jni) [inline doc/examples/registration/plot_register_rotation.py:280]: These could get DOI links — there's examples of this through the docs, something like `DOI`:tip10.34904

**Comment by jni** [comment]
@mellertd sorry, busy times! This is fine, just suggested a typo fix and *maybe* adding DOIs if you get a chance. Otherwise great PR, apologies for the silence! =)

File: doc/examples/registration/plot_register_rotation.py (hunk 2 above)
Comment (mellertd) [reply doc/examples/registration/plot_register_rotation.py]: Took care of those DOIs

**Pull request merged**